omod/src/main/java/org/openmrs/module/referenceapplication/page/controller/LoginPageController.java

```diff
@@ -277,7 +316,8 @@ public String post(@RequestParam(value = "username", required = false) String us
 		//TODO limit login attempts by IP Address
 
 		pageRequest.getSession().setAttribute(SESSION_ATTRIBUTE_REDIRECT_URL, redirectUrl);
-
+		// Since the user is already authenticated without location, need to logout before redirecting
+		Context.logout();
```

Comment:
Since I need to do the context.authenticate() before verifying the location restrictions. So if anything required isn't satisfied, then it will be redirected with the authenticated user. So the user should automatically log out from the system before the redirect.